File: twosum.js
var twoSum = function(nums, target) {
    let hashmap = {}; // To store numbers and their indices
    for (let i = 0; i < nums.length; i++) {
        let complement = target - nums[i];
        if (hashmap.hasOwnProperty(complement)) {
            return [hashmap[complement], i];
        }
        hashmap[nums[i]] = i;
    }
};

File: twosum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}  # To store numbers and their indices
        for i, num in enumerate(nums):
            complement = target - num
            if complement in hashmap:
                return [hashmap[complement], i]
            hashmap[num] = i
